### DISCUSSION TITLE
Analog joystick for crankshaft

### DISCUSSION THREAD

**Discussion by krupsiak** [opencardev/crankshaft#492] [Ideas]
Hello guys,
I have an idea to use an analog joystick:
example:
https://www.makerfabs.com/analog-2-axis-thumb-joystick-with-select-button.html

With raspberry we can use analog joy. 
https://www.tomshardware.com/how-to/raspberry-pi-pico-joystick

I would like to use joy as left,right,up,down and button as enter.
How to integrate it?

**Comment by matt2005** [comment]
Use gpio2kbd.cfg I'll add a wiki page in a bit

**Reply by krupsiak** [reply]
Hi, thx for a fast reply. I understand that I should bind analog signal to keyboard key. But I don't know how to do it exactly. Should I first initialize script with translating analog axis signal to digital value?

**Reply by matt2005** [reply]
If you look in to gpio2kbd.cfg in the boot partition. It has gpio to keyboard mapping. You will just need connect and configure it. You will then need to enable the keyboard mapping in crankshaft config.
